src/model/index.ts

```diff
@@ -215,13 +250,66 @@ class Model implements WorkbookModel {
       ids.add(value.security_id);
       this.issuedSecuritiesByStakeholderAndStockClassIds_.set(key, ids);
     }
+    const txns =
+      this.transactionsBySecurityId_.get(value.security_id) || new Set();
+    txns.add(value);
+    this.transactionsBySecurityId_.set(value.security_id, txns);
+  }
+
+  // eslint-disable-next-line @typescript-eslint/no-explicit-any
+  private async TX_PLAN(value: any) {
+    if (value.object_type === "TX_PLAN_SECURITY_ISSUANCE") {
+      const key = `${value.stakeholder_id}/${value.stock_plan_id}`;
+      const ids =
+        this.issuedSecuritiesByStakeholderAndStockPlanIds_.get(key) ||
+        new Set();
+      ids.add(value.security_id);
+      this.issuedSecuritiesByStakeholderAndStockPlanIds_.set(key, ids);
+    }
 
+    if (value.object_type === "TX_PLAN_SECURITY_RETRACTION") {
+      // find the issuance (value of type TX_PLAN_SECURITY_ISSUANCE in transactionsBySecurityId_) by value.security_id
+      // If we don’t have the issuance yet, then we have to hold on to the value.security_id until the issuance is loaded
+      const issuanceTxn = Array.from(
+        this.transactionsBySecurityId_.get(value.security_id) || []
+      ).find((txn) => txn.object_type === "TX_PLAN_SECURITY_ISSUANCE");
+      if (issuanceTxn) {
+        value.quantity = issuanceTxn.quantity;
+        value.stock_plan_id = issuanceTxn.stock_plan_id;
+        value.stakeholder_id = issuanceTxn.stakeholder_id;
+        const key = `${value.stakeholder_id}/${value.stock_plan_id}`;
+        const ids =
+          this.issuedSecuritiesByStakeholderAndStockPlanIds_.get(key) ||
+          new Set();
+        ids.add(value.security_id);
+        this.issuedSecuritiesByStakeholderAndStockPlanIds_.set(key, ids);
+        this.checkPendingTransactions(value);
+      } else {
+        this.pendingTransactionsBySecurityId_.push(value);
+      }
+    }
```

Comment:
**Question:** I don't understand all this complexity, and inconsistency with the existing approach. The existing calculator handled `TX_STOCK_REISSUANCE`, which behaves similarly, by pushing all reissuance onto a list and then resolving them in `get value()`. The same could be done with retractions, couldn't it?